### ISSUE TITLE
[ALM Accelerator - BUG]: The Solution Branch is not being created for the new repository

### ISSUE THREAD

**Issue by Srikanth-Singapuram** [microsoft/coe-starter-kit#8875]
### Describe the issue

The Solution Branch is not being created for the new repository because of this the export pipeline is not establishing build policies. However, when I utilize the default repository to create a new feature, the new solution branch is successfully generated. I am uncertain about the underlying cause of this issue.

I have followed the same [steps ](https://github.com/microsoft/coe-starter-kit/issues/7057#issuecomment-1807753131)

![image](https://github.com/user-attachments/assets/5726e341-9f86-4a5f-85de-1d5f14a8d201)

### What component are you experiencing the issue with?

ALM Accelerator Canvas App

### What solution version are you using?

1.0.20240305.1

**Comment by mikefactorial** [comment]
@Srikanth-Singapuram there's usually a message in the pipeline run log that would indicate what the error is in cases like this somewhere in the Update Deployment Settings step. It won't fail but an error will be logged. There was an update to Azure DevOps this year that requires additional permissions to be set that I don't believe are documented in the ALM Setup docs that affected how the Build Policy is created that you may be running into. Check here https://github.com/microsoft/coe-starter-kit/issues/7930#issuecomment-2024283993

**Comment by mikefactorial** [comment]
@RajeevPentyala I made an update for this a while back in the Project setup wizard in this PR https://github.com/microsoft/coe-starter-kit/pull/7959/files#r1734540303. Not sure if that will ever get added to a release but at the very least should update the docs to include this new required permission here https://github.com/MicrosoftDocs/power-platform/blob/main/power-platform/guidance/alm-accelerator/setup-components-manually.md#set-permissions-for-the-project-build-service

**Comment by Srikanth-Singapuram** [comment]
@mikefactorial I appreciate your prompt reply. I did not identify any errors related to the failure in creating the solution branch. However, I observed the following discrepancy, which may be the cause of the issue due to a failing condition. The log on the left side indicates the solution branch created from the default repository, while the log on the right side shows the failure to create the solution branch for the new repository. We possess the "Queue builds" permission for the pipeline "export-solution-to-git."

![image](https://github.com/user-attachments/assets/95199d32-084f-4dc6-bb4d-264e2f8bed68)

![image](https://github.com/user-attachments/assets/b0547edc-286a-45d6-aa07-44b83bf5f74b)

![image](https://github.com/user-attachments/assets/b0eb434a-7890-4bfc-a103-812bb7399059)

**Comment by mikefactorial** [comment]
@Srikanth-Singapuram thanks for the additional details. There is an assumption, unfortunately. in the pipelines that the default branch is called "main". Looking back at your original screenshot I see now that the default in this case is "master" that's likely the cause of the issue.

**Comment by Srikanth-Singapuram** [comment]
@mikefactorial Mike, after granting the "Queue builds" permission for the pipeline, I am encountering the following error with the new repository.  I believe It consistently references the default repository, which shares the same name as the ADO project, and the GUID of that repository in the error corresponds to the default repository.

@RajeevPentyala 

![image](https://github.com/user-attachments/assets/a28d034c-2988-4fa9-bcad-59a6e7ff001e)

![image](https://github.com/user-attachments/assets/8e55c263-c55c-4661-aaf3-a738dba67202)

![image](https://github.com/user-attachments/assets/8c302cab-1c72-4bfb-b758-148fdd67e3a7)

**Comment by Srikanth-Singapuram** [comment]
@mikefactorial  Indeed, you are correct. In the PowerShell script, we are evaluating the condition "**if($refBranch.name -eq 'refs/heads/main')**". Instead, we could incorporate an "or" condition to include the value **"refs/heads/master"**.

![image](https://github.com/user-attachments/assets/11dae991-f849-4132-b444-7c6336a588b4)


When we initiate the REST API request for the Default Repository.
![image](https://github.com/user-attachments/assets/8fe5407a-00d5-4931-9741-6b48ac9ca014)

When we initiate the REST API request for the new Repository.
![image](https://github.com/user-attachments/assets/c99f2c23-80f7-43b4-957c-f206c30710bb)

**Comment by AmarSaiRam** [comment]
Hi, According to https://learn.microsoft.com/en-us/power-platform/alm/pipelines, Power Platform Pipelines are now the recommended approach for Application Lifecycle Management (ALM). The ALM Accelerator is no longer actively maintained or supported. Therefore, this issue is being closed accordingly.
